clients/pubsub-common/src/main/java/org/eclipse/hono/client/pubsub/PubSubBasedAdminClientManager.java

```diff
@@ -148,68 +145,53 @@ public Future<Void> closeAdminClients() {
             return Future.succeededFuture();
         }
         final Promise<Void> adminClientClosePromise = Promise.promise();
-        if (context == null) {
-            throw new IllegalStateException("Clients are not running on a Vert.x Context");
-        } else {
-            context.executeBlocking(future -> {
-                closeSubscriptionAdminClient();
-                closeTopicAdminClient();
-                future.complete();
-            }, adminClientClosePromise);
-            return adminClientClosePromise.future();
-        }
+        final Context context = vertx.getOrCreateContext();
+        context.executeBlocking(future -> {
+            closeSubscriptionAdminClient();
+            closeTopicAdminClient();
+            future.complete();
+        }, adminClientClosePromise);
+        return adminClientClosePromise.future();
     }
 
     private Future<Void> getOrCreateTopic(final TopicName topic) {
+        if (topics.contains(topic.toString())) {
+            LOG.debug("Topic {} already exists, continue", topic);
+            return Future.succeededFuture();
+        }
         return topicAdminClientCreator.get()
-                .onFailure(thr -> LOG.error("admin client creation failed", thr))
-                .recover(Future::failedFuture)
+                .onFailure(thr -> LOG.debug("admin client creation failed", thr))
```

Comment:
it feels a little strange to keep the reference to the *createTopicAdminClient()* method in a field instead of simply invoking the private *createTopicAdminClient()* method directly and pass its result into the *getOrCreateTopic()* method on success ...

Reply:
The createTopicAdminClient() method is invoked in the constructor and set to the supplier topicAdminClientCreator, because the initialization of the TopicAdminClient might take some time. So we don't think we should call the createTopicAdminClient() method in here..

Reply:
Can you help me by pointing out the location where the *createTopicAdminClient()* method is being invoked in the constructor? I can only see the assignment:
```java
Supplier<Future<TopicAdminClient>> topicAdminClientCreator = this::createTopicAdminClient;
```
which, IMHO, simply keeps a reference to the actual method instead of invoking it. Or am I mistaken?

Reply:
Since Michelle won't have the time to follow up on this PR, she asked me to take it over. From my point of view you are right, so I removed the creator parameters and instead invoked the method directly.

Reply:
IMHO it does not make much sense to store the admin client in the *topicAdminClient* field if the field is never being read ....

Reply:
good point, I changed the implementation to make use of the _topicAdminClient_ in a meaningful way.